tests/flows/fulfillment/test_shared.py

```diff
@@ -588,6 +590,74 @@ def test_set_or_update_coterm_next_sync_dates_step_are_in_sync(
     mocked_next_step.assert_called_once_with(mocked_client, context)
 
 
+def test_set_or_update_coterm_next_sync_dates_step_with_3yc(
+    mocker,
+    order_factory,
+    adobe_customer_factory,
+    adobe_commitment_factory,
+):
+    """
+    Tests that the order is updated with 3YC parameters when the Adobe customer
+    has a 3YC commitment request.
+    """
+    mocked_update = mocker.patch("adobe_vipm.flows.fulfillment.shared.update_order")
+    commitment = adobe_commitment_factory(
+        status="REQUESTED",
+        start_date="2024-01-01",
+        end_date="2027-01-01",
+    )
+    customer = adobe_customer_factory(
+        coterm_date="2025-01-01",
+        commitment_request=commitment
+    )
+
+    order = order_factory()
+
+    context = Context(
+        order=order,
+        order_id=order["id"],
+        adobe_customer_id=customer["customerId"],
+        adobe_customer=customer,
+    )
+
+    mocked_client = mocker.MagicMock()
+    mocked_next_step = mocker.MagicMock()
+
+    step = SetOrUpdateCotermNextSyncDates()
+    step(mocked_client, context, mocked_next_step)
+    mocked_update.assert_called_once_with(
+        mocked_client,
+        context.order_id,
+        parameters=context.order["parameters"],
+    )
+
+    assert get_fulfillment_parameter(
```

Comment:
Please try to have minimum number of asserts in the test. It is better to compare whole list of parameters here